### PR TITLE
Fixes #378 #504 #517: Media Wall

### LINKED ISSUE #378: Remove unnecessary imports from the modules
There are many unnecessary imports in the all the modules.
Remove them so that code becomes cleaner and maintainable

### PR DESCRIPTION
**Pull request by Achint08**
**Changes proposed in this pull request**
- Improves Media Wall UI/UX
- Adds various Customization Options

**Follow Up Issues**
- Adding Direct URL
- Adding Candy Card Style
- Adding Lightbox for card

**Link to live demo (if appropriate):** 
https://achint08.github.io/loklak_search/wall?query=%23trump

**Closes #378 #504 #517 **

### CODE REVIEW

File: src/app/actions/media-wall-custom.ts [hunk 1]
```diff
@@ -12,7 +12,9 @@ import { WallHeader, WallBackground, WallCard } from '../models';
 export const ActionTypes = {
 	WALL_HEADER_PROPERTIES_CHANGE: '[Media Wall] Media Wall Header Properties Change',
 	WALL_BACKGROUND_PROPERTIES_CHANGE: '[Media Wall] Media Wall Background Properties Change',
-	WALL_CARD_PROPERTIES_CHANGE: '[Media Wall] Media Wall Card Properties Change'
+	WALL_CARD_PROPERTIES_CHANGE: '[Media Wall] Media Wall Card Properties Change',
+	WALL_LIGHT_THEME_CHANGE: '[Media Wall] Media Wall Basic Theme',
```
Comment (Achint08) [inline src/app/actions/media-wall-custom.ts]: Added Actions for Themes here.

File: src/app/actions/media-wall-design.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * For each action type in an action group, make a simple
+ * enum object for all of this group's action types.
+ *
+ * The 'actionTypeCheck' utility function coerces strings into string
+ * literal types and runs a simple check to guarantee all
+ * action types in the application are unique.
+ */
+export const ActionTypes = {
+	WALL_DISPLAY_HEADER_ACTION: '[Media Wall] Media Wall Display Header',
```
Comment (Achint08) [inline src/app/actions/media-wall-design.ts]: Actions for media wall design here.

File: src/app/effects/api-search.effects.ts [hunk 3]
```diff
@@ -174,6 +174,16 @@ export class ApiSearchEffects {
 							return new wallPaginationAction.WallNextPageAction('');
 					});
 
+	@Effect()
+	nextWallSearchActionAfterFail$
```
Comment (Achint08) [inline src/app/effects/api-search.effects.ts]: Now, if there is no response from API, next request in 5 seconds.

File: src/app/models/media-wall-filters.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
+import { ApiResponseResult } from '../models/api-response';
```
Comment (Achint08) [inline src/app/models/media-wall-filters.ts]: All the functions for filters here.

**Comment by Achint08** [comment]
Please review @hemantjadon @singhpratyush @vibhcool @djmgit  @kavithaenair @SKrPl .

**Comment by djmgit** [comment]
aa2

**Comment by djmgit** [comment]
@Achint08 please resolve conflict and rebase.

**Comment by Achint08** [comment]
@hemantjadon @djmgit Done with the request changes.

**Comment by hemantjadon** [comment]
@Achint08 can we have in image cards full width images, there is a minor margin around the images. Remove it in my opinion

**Comment by Achint08** [comment]
@hemantjadon I started with that. It wasn't looking good that way.

**Comment by Achint08** [comment]
@hemantjadon Done!

**Comment by Achint08** [comment]
@singhpratyush 
> Search term at the bottom appears on abrupt location, not aligned with anything.

Just wait for other options to be added. It will be aligned

> Direct URL does nothing for me.

Please see the description. By next PR, it will work.

**Comment by Achint08** [comment]
@singhpryush 
> Not enough padding for Color Options.

Done!

**Pull request merged**